src/scriptworker/utils.py

```diff
@@ -217,7 +218,15 @@ def calculate_sleep_time(attempt, delay_factor=5.0, randomization_factor=0.5, ma
 
 
 # retry_async {{{1
-async def retry_async(func, attempts=5, sleeptime_callback=calculate_sleep_time, retry_exceptions=Exception, args=(), kwargs=None, sleeptime_kwargs=None):
+async def retry_async(
+    func: Callable[..., Awaitable[Any]],
+    attempts: int = 5,
+    sleeptime_callback: Callable[..., Any] = calculate_sleep_time,
+    retry_exceptions: Union[Type[BaseException], Tuple[Type[BaseException], ...]] = Exception,
+    args: Sequence[Any] = (),
+    kwargs: Optional[Dict[str, Any]] = None,
+    sleeptime_kwargs: Optional[Dict[str, Any]] = None,
+) -> Any:
```

Comment:
Cool, good to see how we can encapsulate complex args and kwargs in typing.
It looks like you're intentionally only covering specific functions? Is there any logic to which?

Reply:
I'm tackling one whole file at a time; and then the functions I'm in related files that I am touching, are ones that are causing errors in the files I am touching -- e.g. `error: Call to untyped function "retry_async" in typed context`